Source/Core/AudioCommon/AudioDevice.cpp

```diff
@@ -0,0 +1,212 @@
+
+#if defined _WIN32
+#include <Mmdeviceapi.h> // Must be included before devpkey:
+#include <Functiondiscoverykeys_devpkey.h>
+#elif defined HAVE_ALSA && HAVE_ALSA
+#include <alsa/asoundlib.h>
+#endif
+
+#include "AudioDevice.h"
+#include "Common/StringUtil.h"
+#include "Core/ConfigManager.h"
+
+std::vector<AudioDevice> AudioDevice::devices;
+// index 0 fits XAudio2_7 default, id "default" fits ALSA default
+AudioDevice AudioDevice::DEFAULT = AudioDevice(0, "Default audio device", "", "default");
+bool AudioDevice::loaded = false;
+
+void AudioDevice::LoadDevices()
```

Comment:
The fact this conditional statement is piled with ifdefs tells you it should be split up via inheritance.